src/main/scala/io/iohk/ethereum/blockchain/sync/FastSync.scala

```diff
@@ -266,10 +276,16 @@ trait FastSync {
       requestedReceipts = requestedReceipts - handler
     }
 
+    private def validateBlocks(requestedHashes: Seq[ByteString], blockBodies: Seq[BlockBody]): Boolean = (requestedHashes zip blockBodies)
```

Comment:
Yes! Given the integrated nature of the response any bad block means we don't trust the sender at all and so we fail the response in it's entirety immediately.